### PR TITLE
Issue #3407056 by nkoporec: Error: Call to a member function get() on null Drupal\\social_activity\\EmailTokenServices->getUserPreview()

### PR DESCRIPTION
**Pull request by nkoporec**
## Problem
When expecting logs we found the following error, popping up multiple times:

Error: Call to a member function get() on null Drupal\\social_activity\\EmailTokenServices->getUserPreview() (line 251 of /app/html/profiles/contrib/social/modules/social_features/social_activity/src/EmailTokenServices.php)

## Solution
Make sure that the field exists before calling get()

## Issue tracker
https://www.drupal.org/project/social/issues/3407056

## Theme issue tracker
*[Required if applicable] Paste a link to the drupal.org theme issue queue item, either from [socialbase](https://www.drupal.org/project/socialbase) or [socialblue](https://www.drupal.org/project/socialblue). If any other issue trackers were used, include links to those too.*

## How to test
- [ ] Install Open Social
- [ ] Create a group  and add group members
- [ ] Send an email to group members and use a token in the email

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
Fixed an undefined error that occurred when sending out emails with tokens.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

File: modules/social_features/social_activity/src/EmailTokenServices.php [hunk 1]
```diff
@@ -247,8 +247,11 @@ public function getUserPreview(User $user) {
     /** @var \Drupal\image\Entity\ImageStyle $image_style */
     $image_style = ImageStyle::load('social_medium');
 
-    /** @var \Drupal\file\FileInterface $image */
-    $image = !$profile->get('field_profile_image')->isEmpty() ? $profile->get('field_profile_image')->entity : '';
+    /** @var \Drupal\file\FileInterface $image|null */
+    $image = NULL;
+    if ($profile->hasField('field_profile_image')) {
+      $image = !$profile->get('field_profile_image')->isEmpty() ? $profile->get('field_profile_image')->entity : '';
```
Comment (vcsvinicius) [inline modules/social_features/social_activity/src/EmailTokenServices.php:253]: This variable should be FileInterface or Nullable, but if profile image is empty will be a string, change it to Nullable

**Review by vcsvinicius — CHANGES_REQUESTED** [review]
@nkoporec I add an assignment

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by nkoporec** [comment]
@vcsvinicius this is ready for re-review

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Review by vcsvinicius — APPROVED** [review]
@nkoporec I approved the PR

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3638-9a4h8tztb2kuekbxocnummttosssevcs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/657332b70430b6e23725781c

**Comment by ronaldtebrake** [comment]
Also cherry-picked to 12.1.x and 12.0.x

**Pull request merged**